### DISCUSSION TITLE
v1.58.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#2587] [Announcements]
# Breaking changes 

The server and the mobile app must be on the same version **(v1.58.0)** for the Partner sharing feature on the mobile app to work correctly.
The `docker-compose.yml` has changed the content. Please make sure to update yours.

![image](https://github.com/immich-app/immich/assets/27055614/359dbe1b-3d8e-471c-94a2-de464da2bcc9)

# Highlights

Hello everyone, I am happy to announce the 102nd release of Immich, which includes the following highlights:

- Map filtering
- You can now view your Partner’s assets on the mobile app and also manage your Partner’s accessibility to your timeline on the mobile app.
- Immich’s server can now understand XMP Sidecars files. 
- We added support for client-side hashing in preparation for the improved upload process across the clients. 
- We improved the stability of the server when handling a large amount of new incoming assets by managing the queues better. 

### Map's filters

You can now use the option button on the map page to filter the display cluster by date-time

![image](https://github.com/immich-app/immich/assets/27055614/26080965-e353-430a-86d9-1692f7c493da)

### Partner sharing on mobile

You can now access your partners’ timelines and manage your shared partners directly on the mobile app, thanks to the work of @fyfrey. The `Share with partner` is no longer grayed out (woohoo).

### Support XMP Sidecar files

Thank @alex-phillips and @jrasm91 contribution to this feature. Immich can now ingest XMP sidecars on file upload (via the CLI) as well as detect new sidecars that are placed in the filesystem for existing images.

XMP sidecars are external XML files that contain metadata related to media files. Many applications read and write these files exclusively or in addition to the metadata written to image files. They can be a powerful tool for editing and storing metadata of a media file without modifying the media file itself. When Immich receives or detects an XMP sidecar for a media file, it will attempt to extract the metadata from both the sidecar and the media file. It will prioritize the metadata for fields in the sidecar but will fall back and use the metadata in the media file if necessary.

When importing files via the CLI bulk uploader, Immich will automatically detect XMP sidecar files that exist next to the original media file and have the exact same name with an additional .xmp file extension (i.e., PXL_20230401_203352928.MP.jpg and PXL_20230401_203352928.MP.jpg.xmp).

![image](https://github.com/immich-app/immich/assets/27055614/8ef76aac-f3b5-419d-b2fc-8a05c24e3130)

Two administrator jobs are associated with sidecar files: **SYNC** and **DISCOVER**. The sync job will re-scan all media with existing sidecar files and queue them for a metadata refresh. This is a great use case when third-party applications are used to modify the metadata of media. The discover job will attempt to scan the filesystem for new sidecar files for all media that does not currently have a sidecar file associated with it.

![image](https://github.com/immich-app/immich/assets/27055614/25347fc5-51f5-4b9d-ad3a-c57bab747fc2)
Please make sure you are using the latest CLI version to be able to upload the XMP file to the server.

**Note: When using external software to manage sidecar metadata, please keep in mind that Immich will not detect changes in the main image file. Please make sure to run the SYNC job to get the data updated**

### Support for client-side hashing

With the contribution of @etnoy, the server now has a mechanism that checks for client-side hashing, which will improve the upload process on the CLI and the mobile app in the upcoming releases.

### Improve queuing system

In one sitting, @jrasm91 single-handedly bent the server to his will; the queuing system now obeys his commands to run the background job orderly without bogging down the CPU. This improvement helps with the server's stability, solves request timeout issues to the database connection, and resolves missing thumbnail situations due to stale data. 

There are many additional changes to improve the application as a whole. Please check out the detailed changelog below.

----

As always, please consider supporting the project.

🎉 Cheer! 🎉

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="support"> 
</p>

If you find the project helpful and it helps you in some ways, you can support the project [one time](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502) or [monthly](https://github.com/sponsors/alextran1502) from GitHub Sponsors

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Mobile
* feat(mobile): partner sharing by @fyfrey in https://github.com/immich-app/immich/pull/2541
* chore(mobile): update isar by @fyfrey in https://github.com/immich-app/immich/pull/2571

### Web
* feat(web): job tile icons by @jrasm91 in https://github.com/immich-app/immich/pull/2546
* feat(web): use user layout on admin pages by @jrasm91 in https://github.com/immich-app/immich/pull/2550
* feat(web): show assets without thumbs by @jrasm91 in https://github.com/immich-app/immich/pull/2561
* feat(web+server): map date filters + small changes by @michelheusschen in https://github.com/immich-app/immich/pull/2565
* fix(web): back button on person page by @michelheusschen in https://github.com/immich-app/immich/pull/2566
* feat(web): improved action bar actions by @jrasm91 in https://github.com/immich-app/immich/pull/2553
* fix(web): small style issues by @michelheusschen in https://github.com/immich-app/immich/pull/2578
* fix(web): hide album context menu by @jrasm91 in https://github.com/immich-app/immich/pull/2543

### Server
* refactor(server): use cascades for keys and tokens by @jrasm91 in https://github.com/immich-app/immich/pull/2544
* fix(server): reverse geocoding delete dump logic by @jrasm91 in https://github.com/immich-app/immich/pull/2551
* refactor(server): album controller by @jrasm91 in https://github.com/immich-app/immich/pull/2539
* feat(server): Add support for client-side hashing by @etnoy in https://github.com/immich-app/immich/pull/2072
* feat(server): xmp sidecar metadata by @alex-phillips in https://github.com/immich-app/immich/pull/2466
* refactor(server): create album by @jrasm91 in https://github.com/immich-app/immich/pull/2555
* refactor(server): update album by @jrasm91 in https://github.com/immich-app/immich/pull/2562
* refactor(server): bull jobs by @jrasm91 in https://github.com/immich-app/immich/pull/2569
* refactor(server): delete album by @jrasm91 in https://github.com/immich-app/immich/pull/2570
* refactor(server): job handlers by @jrasm91 in https://github.com/immich-app/immich/pull/2572
* feat(server): disk stats for library folder by @jrasm91 in https://github.com/immich-app/immich/pull/2560
* fix(server): invalid exif date string by @jrasm91 in https://github.com/immich-app/immich/pull/2580
* fix(server): exif time extraction by @michelheusschen in https://github.com/immich-app/immich/pull/2583
* fix(server): missing metadata extraction job by @alextran1502 in https://github.com/immich-app/immich/pull/2586

### Other
* docker: use default entry point and command where applicable by @uhthomas in https://github.com/immich-app/immich/pull/2529

## New Contributors
* @alex-phillips made their first contribution in https://github.com/immich-app/immich/pull/2466

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.57.1...v1.58.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.58.0'>v1.58.0</a>.</em>

**Comment by johannesparty** [comment]
Apple Photos export (Latest Sonoma) puts .xmp next to image file, like "abc.jpg" and "abc.xmp".  Is there a way to make immich find these files too?

**Reply by etnoy** [reply]
Not yet, but it's on my todo list
